librz/util/big-ssl.c: fails to cross-compile for ARM targets

While updating rizin to v0.5.1 in Void Linux, came across the following build error in cross-compilation for aarch64 and other ARM targets (truncated to only show errors):

```
FAILED: librz/util/librz_util_native.a.p/big-ssl.c.o 
cc -Ilibrz/util/librz_util_native.a.p -Ilibrz/util -I../librz/util -I. -I.. -Ilibrz -I../librz -Ilibrz/include -I../librz/include -Ilibrz/util/sdb/src -I../librz/util/sdb/src -Ilibrz/util/sdb -I../librz/util/sdb -flto=auto -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -D_GNU_SOURCE -O2 -pipe -fPIC -pthread -MD -MQ librz/util/librz_util_native.a.p/big-ssl.c.o -MF librz/util/librz_util_native.a.p/big-ssl.c.o.d -o librz/util/librz_util_native.a.p/big-ssl.c.o -c ../librz/util/big-ssl.c
../librz/util/big-ssl.c: In function 'rz_big_to_int':
../librz/util/big-ssl.c:33:9: error: unknown type name 'BN_ULONG'
   33 |         BN_ULONG maxx = 0;
      |         ^~~~~~~~
../librz/util/big-ssl.c:35:9: error: unknown type name 'BN_ULONG'
   35 |         BN_ULONG res = BN_get_word(b);
      |         ^~~~~~~~
../librz/util/big-ssl.c:39:33: error: 'BN_BYTES' undeclared (first use in this function); did you mean 'BITS2BYTES'?
   39 |                 BN_mask_bits(B, BN_BYTES * 8 - 1);
      |                                 ^~~~~~~~
      |                                 BITS2BYTES
../librz/util/big-ssl.c: In function 'rz_big_mul':
../librz/util/big-ssl.c:86:9: error: unknown type name 'BN_CTX'
   86 |         BN_CTX *bn_ctx = BN_CTX_new();
      |         ^~~~~~
../librz/util/big-ssl.c: In function 'rz_big_div':
../librz/util/big-ssl.c:92:9: error: unknown type name 'BN_CTX'
   92 |         BN_CTX *bn_ctx = BN_CTX_new();
      |         ^~~~~~
../librz/util/big-ssl.c: In function 'rz_big_mod':
../librz/util/big-ssl.c:98:9: error: unknown type name 'BN_CTX'
   98 |         BN_CTX *bn_ctx = BN_CTX_new();
      |         ^~~~~~
../librz/util/big-ssl.c: In function 'rz_big_divmod':
../librz/util/big-ssl.c:104:9: error: unknown type name 'BN_CTX'
  104 |         BN_CTX *bn_ctx = BN_CTX_new();
      |         ^~~~~~
../librz/util/big-ssl.c: In function 'rz_big_powm':
../librz/util/big-ssl.c:227:9: error: unknown type name 'BN_CTX'
  227 |         BN_CTX *bn_ctx = BN_CTX_new();
      |         ^~~~~~
```

The full build log can be found here: [rizin-log.txt](https://github.com/rizinorg/rizin/files/10821909/rizin-log.txt)
Any pointers for what might be the reason for this?

--- a/librz/util/meson.build
+++ b/librz/util/meson.build
@@ -1,6 +1,6 @@
 subdir('sdb/src')
 
-rz_util_sources = [
+rz_util_common_sources = [
   'alloc.c',
   'annotated_code.c',
   'ascii_table.c',
@@ -83,6 +83,7 @@ rz_util_sources = [
   'w32-sys.c',
   'x509.c',
 ]
+rz_util_sources = rz_util_common_sources
 
 rz_util_deps = [ldl, lrt, mth, th, utl] + platform_deps
 if zlib_dep.found()
@@ -162,8 +163,9 @@ if meson.is_cross_build()
     install_dir: rizin_incdir
   )
   rz_util_native_includes = [include_directories('.'), rz_util_includes]
+  rz_util_native_sources = rz_util_common_sources + ['big.c']
 
-  rz_util_native = static_library('rz_util_native', rz_util_sources, libsdb_sources,
+  rz_util_native = static_library('rz_util_native', rz_util_native_sources, libsdb_sources,
     include_directories: rz_util_native_includes,
     dependencies: rz_util_native_deps,
     implicit_include_directories: false,

